### PR TITLE
Generate fake data for backend testing.

### PR DESCRIPTION
**Pull request by JacksonStaniec**
### Summary 

This PR adds `createData.js` that uses Faker.js to create mock `IdolMembers`, `TeamEventAttendance`, and `TeamEvent` objects for the backend test suite. Hardcoded mock data is replaced by generated mock data wherever possible.

### Test Plan 
Test cases behave as expected.

### CODE REVIEW

File: backend/tests/MembersDao.test.ts [hunk 1]
```diff
@@ -26,8 +31,8 @@ test('Add new member', () => {
 });
 
 test('Get member from past semester', () =>
-  MembersDao.getCurrentOrPastMemberByEmail(jaggerData.email).then((pastMember) =>
-    expect(pastMember).toEqual(jaggerData)
+  MembersDao.getCurrentOrPastMemberByEmail(jaggerData.email).then(
+    (pastMember) => expect(pastMember).toEqual(jaggerData) // cannot deprecate jaggerData yet
```
Comment (JacksonStaniec) [inline backend/tests/MembersDao.test.ts:35]: This is the only place left that uses the hardcoded mock user `jaggerData`. This test for the `getCurrentOrPastMemberByEmail` function relies on accessing a json archive of previous members. Is it worth it to automatically add a generated user to the archive and remove it after the test runs, or is it fine to keep the hardcoded mock data in this case?

**Review by henry-li-06 — COMMENTED** [review]
did you get a chance to look into fixing `update existing team event with permission` test

**Comment by JacksonStaniec** [comment]
> did you get a chance to look into fixing `update existing team event with permission` test

It seems like since the Idol member in `TeamEventAttendance` objects is stored in Firestore as a document reference to the members collection, and since we have been using fake members, its failing because there is no actual member in the database and so the retrieved object from the database has `member: undefined`. This is probably the case for `TeamEventsDao` tests too. We could either:
- first put the fake Idol member in the database, and remove it afterwards like what happens in the `MembersDao` tests
- compare specific fields of the expected and received object to check for equality, instead of the objects themselves.

**Comment by henry-li-06** [comment]
> * database

I think the first option seems pretty reasonable

**Review by henry-li-06 — APPROVED** [review]
let's forget about fixing that last test for now

**Pull request merged**